src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -52,16 +54,21 @@ private async Task LoadAsync()
             var secrets = await _client.GetSecretsAsync(_vault).ConfigureAwait(false);
             do
             {
+                var tasks = new List<Task<KeyValuePair<string, string>>>();
```

Comment:
Can the List be initialized with the count of `secrets`?
Though in this case it's really minor...